### ISSUE TITLE
File-list datatype fails to index when importing from csv

### ISSUE THREAD

**Issue by ryan86** [archesproject/arches#4044]
When importing a csv that was exported from Arches that includes a file-list datatype indexing of the resource fails with the following error:

`WARNING: failed to index document in resource: d49dd794-1a5a-11e7-af19-6c4008b05c4c on line 13. Exception detail: 'str' object has no attribute 'FILES'`

**Comment by mradamcox** [comment]
Hey @ryan86 to save you a little investigative work, I encountered this last week and figured out that this is because the upload function is attempting to run, which wants to process a request object. If you `import_business_data ... --bulk` that function doesn't run, so that was my temporary work around for the time being.

**Comment by ryan86** [comment]
@mradamcox Thanks!

**Comment by jmunowitch** [comment]
This looks to be caused by having a json object in the csv, without accounting for the extra commas.
```
"[{u'status': u'uploaded', u'index': 0, u'name': u'test.jpeg', u'url': u'/files/uploadedfiles/test.jpeg', u'lastModified': 1513803988000, u'height': 652, u'content': u'blob:http://localhost:8000/71ba326c-3994-794f-a015-3470772ff768', u'width': 960, u'file_id': u'7250258f-abb0-11e8-814a-b8f6b115d7dd', u'accepted': True, u'type': u'image/jpeg', u'size': 147133}]"
```

Exporting the resource to JSON and importing worked.

**Comment by beatboxchad** [comment]
This issue cropped up in our way as we worked on the v3-v4 migration project, and we were thus inspired to attain clarity.

tl;dr @mradamcox is right. It's not having to do with any data within the CSV.

Rather, it has to do with  [_arches/app/utils/data_management/resources/formats/csvfile.py_](https://github.com/archesproject/arches/blob/55377b5753039b56cda8019f793e38b71e1b522a/arches/app/utils/data_management/resources/formats/csvfile.py#L295-L311) calling `Resource.save` without a request object. Since this is called from a management command, there is no request object. The error bubbles up from `Tile.check_for_missing_nodes` within `Tile.save`, but `Tile.save` gracefully defaults to `None` for the request and only runs such view code `if request is not None:`. Surely `Resource.save` should do the same thing! As it turns out, `Resource.save`  coerces an absent request to an [empty string](https://github.com/archesproject/arches/blob/55377b5753039b56cda8019f793e38b71e1b522a/arches/app/models/resource.py#L89-L97), which technically counts as data, and passes that to its `Tile`s for their use in saving.

So that's the bug. PR coming soon.

**Comment by ryan86** [comment]
meets requirements.
